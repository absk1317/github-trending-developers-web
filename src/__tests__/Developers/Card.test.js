import React from 'react';
import { create } from 'react-test-renderer';
import DeveloperCard from '../../views/Developers/Card';

const developer = {
  username: 'SomeUserName',
  name: 'Some Name',
  type: 'User',
  url: 'some url',
  avatar: 'some avatar',
  repo: {
    name: 'RepoName',
    description: 'Repo Description',
    url: 'RepoURL',
  },
};

// specs for Card Component for matching the snapshot
describe('DeveloperCard Component', () => {
  test('Matches the snapshot', () => {
    const card = create(<DeveloperCard developer={developer} />);
    expect(card.toJSON()).toMatchSnapshot();
  });
});
