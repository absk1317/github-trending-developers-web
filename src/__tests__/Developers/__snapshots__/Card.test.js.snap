// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeveloperCard Component Matches the snapshot 1`] = `
<div
  className="CardContainer card"
>
  <a
    href="some url"
    rel="noopener noreferrer"
    style={
      Object {
        "textDecoration": "none",
      }
    }
    target="_blank"
  >
    <img
      className="card-img ProfileIcon"
      src="some avatar"
    />
  </a>
  <div
    className="DeveloperDetails card"
  >
    <div
      className="card-body"
    >
      <div
        className="DeveloperDetailsTitle card-title h5"
      >
        Some Name
        <span
          className="Username"
        >
           (
          SomeUserName
          )
        </span>
        <span
          className="Username"
        >
           (
          User
          )
        </span>
      </div>
      <div
        className="list-group-item"
        disabled={false}
        onClick={[Function]}
      >
        <div
          className="RepoDetails card"
        >
          <div
            className="card-body"
          >
            <button
              className="RepoLink btn btn-primary"
              disabled={false}
              type="button"
            >
              <a
                href="RepoURL"
                rel="noopener noreferrer"
                style={
                  Object {
                    "textDecoration": "none",
                  }
                }
                target="_blank"
              >
                RepoName
              </a>
            </button>
            <div
              className="RepoDescription list-group-item"
              disabled={false}
              onClick={[Function]}
            >
              Repo Description
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
